numhask/src/NumHask/Algebra/Abstract/Multiplication.hs

```diff
@@ -0,0 +1,326 @@
+{-# OPTIONS_GHC -Wall #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ConstrainedClassMethods #-}
+{-# LANGUAGE RoleAnnotations #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE MonoLocalBinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FunctionalDependencies #-}
+-- | The Group hirarchy
+module NumHask.Algebra.Abstract.Multiplication
+      ( one
+      , recip
+      , Mult(..)
+      , coerceFM
+      , coerceFM'
+      , coerceTM
+      , coerceTM'
+      , Multiplication(..)
+      , (*)
+      , zero'
+      , (/)
+      , times
+      )
+      where
+
+import           NumHask.Algebra.Abstract.Group
+import           NumHask.Algebra.Abstract.Addition
+import qualified Prelude                       as P
+import qualified GHC.Generics                  as P
+import Data.Coerce
+import           Data.Complex                   ( Complex(..) )
+import           Data.Int                       ( Int8
+                                                , Int16
+                                                , Int32
+                                                , Int64
+                                                )
+import           Data.Word                      ( Word
+                                                , Word8
+                                                , Word16
+                                                , Word32
+                                                , Word64
+                                                )
+import           GHC.Natural                    ( Natural(..) )
+
+newtype Mult a = Mult a
```

Comment:
Same as for `Add`. This should really be called `Product` for consistency with the rest of base